app/src/main/java/com/example/drawingactivity/MainCanvas.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Rect
+import android.util.AttributeSet
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class MainCanvas @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    // adjust this with primary color from viewModel
+    var pathColor = Color.BLACK
+
+    // boolean to toggle the ability to draw on the canvas by picking a tool
+    var enableDrawing = false
+
+    // bitmap to draw on
+    var bitmap: Bitmap? = null
+    var shape: Rect? = null
+
+    private var pathPaint = Paint().apply {
+
+    }
+
+    private var bitmapPaint = Paint().apply {
+        color = Color.WHITE
+    }
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
```

Comment:
Combine blocks if they're not doing to do anything else

Reply:
see above